smart-contracts/contracts/Unlock.sol

```diff
@@ -272,6 +272,36 @@ contract Unlock is
     return newLock;
   }
 
+  /**
+   * Create an upgradeable lock using a specific PublicLock version
+   * @param data bytes containing the call to initialize the lock template
+   * (refer to createUpgradeableLock for more details)
+   * @param _lockVersion the version of the lock to use
+  */
+  function createUpgradeableLockAtVersion(
```

Comment:
Looks good to me but can we refactor `createUpgradeableLock` to use that function under the hood to make things a bit more DRY please :)

Reply:
yep [14d6b2f](https://github.com/unlock-protocol/unlock/pull/8338/commits/14d6b2f90c84068eae58527b56bd189f39fdb95c)